pkg/controller/kibana/config/settings.go

```diff
@@ -42,7 +54,11 @@ func NewConfigSettings(client k8s.Client, kb kbv1.Kibana, versionSpecificCfg *se
 	kibanaTLSCfg := settings.MustCanonicalConfig(kibanaTLSSettings(kb))
 
 	if !kb.RequiresAssociation() {
-		if err := cfg.MergeWith(versionSpecificCfg, kibanaTLSCfg, userSettings); err != nil {
+		if err := cfg.MergeWith(
```

Comment:
Can the following happen?
1. user sets some `foo:bar` setting in their Kibana manifest
2. that setting is now part of the secret
3. they remove it from their Kibana manifest
4. we still inherit it because we merge from the existing secret

Reply:
Yes, but that seems okay? Deleting the secret will work to re-generate the secret with a new encryption key, so there at least is a way to do that if that is the goal.

Reply:
What I meant was this also apply to any other settings.
If I set:

```yaml
apiVersion: kibana.k8s.elastic.co/v1
kind: Kibana
metadata:
  name: kibana-sample
spec:
  version: 7.5.0
  count: 1
  elasticsearchRef:
    name: "elasticsearch-sample"
  config:
    logging.verbose: true # temporarily enable verbose logging
```

Then 1 hour later once I'm done with looking at Kibana logs I remove my extra config and apply that yaml:

```yaml
apiVersion: kibana.k8s.elastic.co/v1
kind: Kibana
metadata:
  name: kibana-sample
spec:
  version: 7.5.0
  count: 1
  elasticsearchRef:
    name: "elasticsearch-sample"
```

I think the code would still merge the new empty user-provided config with the current config which does have `logging.verbose: true` and that's not what I would expect.

Please correct me if I'm wrong, I may be misreading the way we do the config merge.

Reply:
Seems like we have two setting types: for some we want to always flow from the resource (like `logging.verbose`) and apply our default to it and for others we want to persist whatever was generated the first time (like the `xpack.security.encryptionKey`).

Should we just get the specific settings that we know should be persisted from the secret?

Reply:
Ah that's a good point and a big miss on my part Seb, good catch.

>Should we just get the specific settings that we know should be persisted from the secret?

I think this makes sense.